domain/pricing.go

```diff
@@ -148,9 +133,79 @@ type PricingUpdateListener interface {
 	OnPricingUpdate(ctx context.Context, height uint64, blockMetaData BlockPoolMetadata, pricesBaseQuoteDenomMap PricesResult, quoteDenom string) error
 }
 
+// PoolLiquidityPricerWorker defines the interface for the pool liquidity pricer worker.
+type PoolLiquidityPricerWorker interface {
+	// Implements PricingUpdateListener
+	PricingUpdateListener
+	// ComputeLiquidityCapitalization computes the capitalization of the liquidity for the given denom
+	// using the total liquidity and the price.
+	// Returs zero if the price is zero or if there is any internal error.
+	// Otherwise, returns the computed liquidity capitalization from total liquidity and price.
+	ComputeLiquidityCapitalization(denom string, totalLiquidity osmomath.Int, price osmomath.BigDec) osmomath.Int
+
+	// RepriceDenomMetadata reprices the token liquidity metadata for the denoms updated within the block.
+	// Returns the updated token metadata.
+	// If there is an update for a denom with a later height than the current height, it is skipped, making this a no-op.
+	// Relies on the blockPriceUpdates to get the price for the denoms.
+	// If the price for denom cannot be fetched, the liquidity capitalization for this denom is set to zero.
+	// The latest update height for this denom is updated on completion.
+	RepriceDenomMetadata(updateHeight uint64, blockPriceUpdates PricesResult, quoteDenom string, blockDenomLiquidityUpdatesMap DenomPoolLiquidityMap) PoolDenomMetaDataMap
+
+	// GetHeightForDenom returns zero if the height is not found or fails to cast it to the return type.
+	GetHeightForDenom(denom string) uint64
```

Comment:
Questions:
1. Does it mean we're keeping track of all the block heights which triggered updates for every denom?

Reply:
No, we keep track of the latest height per denom.

I will rename to `GetLatestUpdateHeightForDenom` to reflect this

Reply:
[03565aa](https://github.com/osmosis-labs/sqs/pull/310/commits/03565aa62b9a520565591071b114e0f7b361009e)